sdk/test/UnitTests/DbCommandInterceptorTests.cs

```diff
@@ -0,0 +1,223 @@
+                XRayOptions = new XRayOptions { CollectSqlQueries = false }
+            };
+            recorder.BeginSegment("test");
+            var interceptor = new DbCommandInterceptor(recorder, collectSqlQueries: true);
+
+            // act            
+            await interceptor.InterceptAsync(() => Task.FromResult(0), _command);
+
+            // assert
+            AssertCollected(recorder);
+            recorder.EndSegment();
+        }
+
+        [TestMethod]
+        public void Intercept_CollectsQueries_When_EnabledGlobally()
+        {
+            // arrange
+            var recorder = new AWSXRayRecorder {
+                XRayOptions = new XRayOptions { CollectSqlQueries = true }
+            };
+            var interceptor = new DbCommandInterceptor(recorder);
+            recorder.BeginSegment("test");
+
+            // act
+            interceptor.Intercept(() => 0, _command);
+
+            // assert
+            AssertCollected(recorder);
+            recorder.EndSegment();
+        }
+
+        [TestMethod]
+        public async Task InterceptAsync_CollectsQueries_When_EnabledGlobally()
+        {
+            // arrange
+            var recorder = new AWSXRayRecorder {
+                XRayOptions = new XRayOptions { CollectSqlQueries = true }
+            };
+            var interceptor = new DbCommandInterceptor(recorder);
+            recorder.BeginSegment("test");
+
+            // act
+            await interceptor.InterceptAsync(() => Task.FromResult(0), _command);
+
+            // assert
+            AssertCollected(recorder);
+            recorder.EndSegment();
+        }
+
+        [TestMethod]
+        public void Intercept_DoesNot_CollectQueries_When_EnabledGlobally_And_DisabledLocally()
+        {
+            // arrange
+            var recorder = new AWSXRayRecorder {
+                XRayOptions = new XRayOptions { CollectSqlQueries = true }
+            };
+            var interceptor = new DbCommandInterceptor(recorder, collectSqlQueries: false);
+            recorder.BeginSegment("test");
+
+            // act
+            interceptor.Intercept(() => 0, _command);
+
+            // assert
+            AssertNotCollected(recorder);
+            recorder.EndSegment();
+        }
+
+        [TestMethod]
+        public async Task InterceptAsync_DoesNot_CollectQueries_When_EnabledGlobally_And_DisabledLocally()
+        {
+            // arrange
+            var recorder = new AWSXRayRecorder {
+                XRayOptions = new XRayOptions { CollectSqlQueries = true }
+            };
+            var interceptor = new DbCommandInterceptor(recorder, collectSqlQueries: false);
+            recorder.BeginSegment("test");
+
+            // act
+            await interceptor.InterceptAsync(() => Task.FromResult(0), _command);
+
+            // assert
+            AssertNotCollected(recorder);
+            recorder.EndSegment();
+        }
+
+        private void AssertNotCollected(AWSXRayRecorder recorder)
```

Comment:
`AssertNotCollected` and `AssertCollected` should also assert rest parameters in `CollectSqlInformation()`.

Reply:
I’ve added them because there aren’t any other tests on this, but in general, I don’t think you should make assertions that lie outside the scope of the test. In an ideal world (this never happens), if something is broken you would have a single test fail that points to exactly what the issue is rather than lots of failing tests. And you don’t want to have to update a bunch of tests if you add a member to the payload. 🤷‍♂️

Reply:
@jaredcnance , agree. For the moment feel free to have a new test which checks rest of the parameters. Or else current version works.